apps/python/ruperts_challenge_app/telemetry_utility.py

```diff
@@ -18,22 +18,73 @@ def set_telemetry(self, telemetry):
             self.telemetry_access_lock.release()
 
     def get_telemetry_data(self, currentLaptime, currentSpline):
+
+        def weighted_interpolation(xs, ys, currentx):
+            '''
+            Returns a weighted linear interpolation at currentx from four
+            sampled points in xy space - two ahead and two behind
+
+            Parameters
+            ----------
+            xs : array
+                an array containing 4 ordered x points
+            ys : array
+                an array containing 4 ordered y points
+            currentx : float
+                the current x position for which a y estimate is required
+
+            Returns
+            -------
+            float
+                the estimated y value for current x
+
+            '''
+            grad12 = (ys[2] - ys[1])/(xs[2] - xs[1])
```

Comment:
I'm a bit concerned about division by zero errors here

Reply:
division by 0 should only occur when a car is not moving - i.e the difference between the spline at time a and time b is 0 - Is there not another way to catch that? would the telemetry even put out those cases since the telemetry is 'indexed' by spline position?

Reply:
Yes good point in fact the spline values are keys in a dictionary so it's impossible for there to be two duplicates. The only scenario I can think of is if the telemetry array is suitably short such that the indices wrap around on themselves but this really shouldn't ever happen so I'm happy to drop this